src/please.go

```diff
@@ -614,6 +621,11 @@ var buildFunctions = map[string]func() bool{
 			query.Roots(state.Graph, opts.Query.Roots.Args.Targets)
 		})
 	},
+	"filter": func() bool {
+		return runQuery(true, opts.Query.Filter.Args.Targets, func(state *core.BuildState) {
```

Comment:
I think you can get away with `false` for the first argument (i.e. you do not need a full dependency parse, just having the definitions of these targets will suffice)